src/inference/dev_api/threading/ie_thread_safe_containers.hpp

```diff
@@ -48,6 +49,27 @@ template <typename T>
 using ThreadSafeQueue = tbb::concurrent_queue<T>;
 template <typename T>
 using ThreadSafeBoundedQueue = tbb::concurrent_bounded_queue<T>;
+template <typename T>
+class ThreadSafeBoundedPriorityQueue {
+public:
+    ThreadSafeBoundedPriorityQueue() = default;
+    bool try_push(T&& value) {
+        if (_capacity) {
+            _pqueue.push(std::move(value));
+        }
+        return _capacity;
```

Comment:
If I understand correctly, `try_push` shall return `true` is push succeeded, right? But there is a chance that 'try_push' pushes something on line 58, but if after that some another thread calls set_capacity(0) - then 'try_push' will return 'false' although task is added to queue successfully

```suggestion
        if (_capacity) {
            _pqueue.push(std::move(value));
            return true;
        }
        return false;
```

Reply:
If `ThreadSafeBoundedPriorityQueue` is just a wrapper, then maybe `return _pqueue.push(std::move(value));` shall be used instead of `_pqueue.push(std::move(value)); return true;`